src/Linux/Native/RunAsUser/src/main.cpp

```diff
@@ -438,8 +434,55 @@ int authenticate_and_run(const picojson::object& json) {
         return err;
     }
 
+    char *user_name = strdup(pw->pw_name);
+    gid_t user_gid = pw->pw_gid;
+    uid_t user_id = pw->pw_uid;
+
+    if (auth_only) {
+        std::string allowed_group(json[RTVS_JSON_MSG_GRP].get<std::string>());
+        if (!allowed_group.empty()) {
+            struct group *gp = getgrnam(allowed_group.c_str());
+            if (!gp) {
+                err = errno;
+                logf(log_verbosity::minimal, "Error [getgrnam]:[%d] %s\n", err, strerror(err));
+                return err;
+            }
+
+            gid_t allowed_gid = gp->gr_gid;
+
+            int ngroups = 1000;
+            gid_t *user_groups = calloc_or_exit<gid_t*>(ngroups, sizeof(gid_t));
```

Comment:
It looks like this is freed immediately in the same block, and no other function takes ownership, so this should probably just be `std::vector<gid_t*>`.